src/Datadog.Sketches/DDSketch.cs

```diff
@@ -0,0 +1,305 @@
+    /// the expected quantile value is 100, the computed quantile value is guaranteed to
+    /// be between 99 and 101. If the expected quantile value is 1000, the computed
+    /// quantile value is guaranteed to be between 990 and 1010.
+    /// DDSketch works by mapping floating-point input values to bins and counting the
+    /// number of values for each bin. The underlying structure that keeps track of bin
+    /// counts is store.
+    /// The memory size of the sketch depends on the range that is covered by the input
+    /// values: the larger that range, the more bins are needed to keep track of the
+    /// input values. As a rough estimate, if working on durations with a relative
+    /// accuracy of 2%, about 2kB (275 bins) are needed to cover values between 1
+    /// millisecond and 1 minute, and about 6kB (802 bins) to cover values between 1
+    /// nanosecond and 1 day.
+    /// The size of the sketch can be have a fail-safe upper-bound by using collapsing
+    /// stores. As shown in
+    /// <see href="http://www.vldb.org/pvldb/vol12/p2195-masson.pdf" />;
+    /// the likelihood of a store collapsing when using the default bound is vanishingly
+    /// small for most data.
+    /// </summary>
+    public class DDSketch
+    {
+        private readonly IIndexMapping _indexMapping;
+        private readonly Store _negativeValueStore;
+        private readonly Store _positiveValueStore;
+        private readonly double _minIndexedValue;
+        private readonly double _maxIndexedValue;
+        private double _zeroCount;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DDSketch"/> class.
+        /// </summary>
+        /// <param name="indexMapping">Map between values and store bins</param>
+        /// <param name="negativeValueStore">Storage for negative values</param>
+        /// <param name="positiveValueStore">Storage for positive values</param>
+        public DDSketch(IIndexMapping indexMapping, Store negativeValueStore, Store positiveValueStore)
+            : this(indexMapping, negativeValueStore, positiveValueStore, 0)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DDSketch"/> class.
+        /// </summary>
+        /// <param name="indexMapping">Map between values and store bins</param>
+        /// <param name="negativeValueStore">Storage for negative values</param>
+        /// <param name="positiveValueStore">Storage for positive values</param>
+        /// <param name="minIndexedValue">The minimum value seen by the sketch</param>
+        public DDSketch(IIndexMapping indexMapping, Store negativeValueStore, Store positiveValueStore, double minIndexedValue)
+            : this(indexMapping, negativeValueStore, positiveValueStore, 0, minIndexedValue)
+        {
+        }
+
+        private DDSketch(IIndexMapping indexMapping, Store negativeValueStore, Store positiveValueStore, double zeroCount, double minIndexedValue)
```

Comment:
nit: probably should swap the order of the final two arguments here:
```suggestion
            : this(indexMapping, negativeValueStore, positiveValueStore, minIndexedValue, 0)
        {
        }

        private DDSketch(IIndexMapping indexMapping, Store negativeValueStore, Store positiveValueStore, double minIndexedValue, double zeroCount)
```